pkg/config/orderer.go

```diff
@@ -45,93 +45,125 @@ type Kafka struct {
 	Brokers []string
 }
 
-// NewOrdererGroup returns the orderer component of the channel configuration
+// NewOrdererGroup returns the orderer component of the channel configuration.
 // It defines parameters of the ordering service about how large blocks should be,
-// how frequently they should be emitted, etc. as well as the organizations of the ordering network
-// It sets the mod_policy of all elements to "Admins"
-// This group is always present in any channel configuration
+// how frequently they should be emitted, etc. as well as the organizations of the ordering network.
+// It sets the mod_policy of all elements to "Admins".
+// This group is always present in any channel configuration.
 func NewOrdererGroup(conf *Orderer, mspConfig *mb.MSPConfig) (*cb.ConfigGroup, error) {
-	var (
-		err               error
-		consensusMetadata []byte
-	)
-
 	ordererGroup := newConfigGroup()
 	ordererGroup.ModPolicy = AdminsPolicyKey
 
-	if err = addOrdererPolicies(ordererGroup, conf.Policies, AdminsPolicyKey); err != nil {
+	// add orderer policies
+	if err := addOrdererPolicies(ordererGroup, conf.Policies, AdminsPolicyKey); err != nil {
 		return nil, fmt.Errorf("failed to add policies: %v", err)
 	}
 
-	err = addValue(ordererGroup, batchSizeValue(
-		conf.BatchSize.MaxMessageCount,
-		conf.BatchSize.AbsoluteMaxBytes,
-		conf.BatchSize.PreferredMaxBytes,
+	// add orderer values
+	err := addOrdererValues(ordererGroup, conf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add orderer values: %v", err)
+	}
+
+	// add orderer groups
+	for _, org := range conf.Organizations {
+		ordererGroup.Groups[org.Name], err = newOrdererOrgGroup(org, mspConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create orderer org group %s: %v", org.Name, err)
+		}
+	}
+
+	return ordererGroup, nil
+}
+
+// newOrdererOrgGroup returns an orderer org component of the channel configuration.
+// It defines the crypto material for the organization (its MSP).
+// It sets the mod_policy of all elements to "Admins".
+func newOrdererOrgGroup(conf *Organization, mspConfig *mb.MSPConfig) (*cb.ConfigGroup, error) {
+	return generateOrgConfigGroup(conf, mspConfig)
```

Comment:
This will be addressed in FAB-17529 along with the other places we create an org group (application/consortium etc). Obviously this function shouldn't exist as is.